File: Intersection of Two Linked List.cpp
Time Complexity-O(n)
Space Complexity-(1)
Did the code run on Leetcode? Yes

class Solution {
public:
    int size1;
    int size2;
    int sizeoflinkedlist(ListNode* a)
    {
        int count=0;
        if(a==NULL)
        {
            return 0;
        }
        while(a!=NULL)
        {
            count++;
            a=a->next;
        }
        return count;
    }
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        size1=sizeoflinkedlist(headA);
        size2=sizeoflinkedlist(headB);
        int diff=size1-size2;
        ListNode* a=headA;
        ListNode* b=headB;
        cout<<size1;
        if(diff>0)
        {
            while(diff!=0)
            {
                a=a->next;
                diff--;
            }
        }
        if(diff<0)
        {
            while(diff!=0)
            {
                b=b->next;
                diff++;
            }
        }
        while(a!=NULL &&b!=NULL)
        {
            if(a==b){
                return a;
                break;
            }
            a=a->next;
            b=b->next;
        }
        return NULL;
    }
};

File: intersectionOfTwoLinkedList.java
// Time complexity - O(n)
// Space complexity - O(1)
//Did this code successfully run on Leetcode : Yes
//Any problem you faced while coding this : No
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        
        
        
        int countA = 0;
        int countB = 0;
        ListNode dummyA = headA;
        ListNode dummyB = headB;
        
        while(dummyA.next != null){
            
            dummyA = dummyA.next;
            countA++;
        }
        while(dummyB.next != null){
            
            dummyB = dummyB.next;
            countB++;
        }
        
        if(countA > countB){
            
            while(countA != countB){
                
                headA = headA.next;
                countA--;
            }
        }
         else if(countA < countB){
            
            while(countA != countB){
                
                headB = headB.next;
                countB--;
            }
        }
        
        while(headA != headB && headA.next != null && headB.next != null){
            
            headA = headA.next;
            headB = headB.next;
        }
        
        if(headA == headB){
            
            return headA;
        }
        
        return null;
    }
}